From 346460284023a3d11e3f5121cb2d6f12ef11512e Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 10 Jul 2014 18:05:50 -0700
Subject: [PATCH 874/974] MIPS:OCTEON: Fix compilation issue with mailbox
 handler.

Call octeon_crash_dump() only when CONFIG_KEXEC is defined.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/cavium-octeon/smp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/mips/cavium-octeon/smp.c b/arch/mips/cavium-octeon/smp.c
index cec7b57..2d7dee4 100644
--- a/arch/mips/cavium-octeon/smp.c
+++ b/arch/mips/cavium-octeon/smp.c
@@ -275,7 +275,9 @@ static void __cpuinit octeon_init_secondary(void)
 
 static irqreturn_t octeon_78xx_smp_dump_interrupt(int irq, void *dev_id)
 {
+#ifdef CONFIG_KEXEC
 	octeon_crash_dump();
+#endif
 	return IRQ_HANDLED;
 }
 
-- 
2.6.2

